crates/editor/src/element.rs

```diff
@@ -1558,6 +1589,258 @@ impl EditorElement {
+
+        elements
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    fn gather_inline_diagnostics(
+        &self,
+        snapshot: &EditorSnapshot,
+        hitbox: &Hitbox,
+        line_layouts: &[LineWithInvisibles],
+        crease_trailers: &[Option<CreaseTrailerLayout>],
+        content_origin: gpui::Point<Pixels>,
+        scroll_pixel_position: gpui::Point<Pixels>,
+        line_height: Pixels,
+        em_width: Pixels,
+        start_row: DisplayRow,
+        end_row: DisplayRow,
+        cx: &mut WindowContext,
+    ) -> HashMap<DisplayRow, DiagnosticElement> {
+        let mut inline_diagnostics: HashMap<DisplayRow, DiagnosticElement> = HashMap::default();
+        let buffer_snapshot = snapshot.display_snapshot.buffer_snapshot.clone();
+
+        let start_point =
+            snapshot.display_point_to_point(DisplayPoint::new(start_row, 0), Bias::Left);
+        let end_point = snapshot.display_point_to_point(DisplayPoint::new(end_row, 0), Bias::Right);
+        let diagnostics = buffer_snapshot
+            .diagnostics_in_range(start_point..end_point, false)
+            .map(|DiagnosticEntry { diagnostic, range }| DiagnosticEntry {
+                diagnostic,
+                range: range.to_point(&buffer_snapshot),
+            })
+            .sorted_by_key(|diagnostic| {
+                (
+                    diagnostic.diagnostic.severity,
+                    std::cmp::Reverse(diagnostic.diagnostic.is_primary),
+                    diagnostic.range.start.row,
+                    diagnostic.range.start.column,
+                )
+            });
+
+        let max_entries = 5; // ToDo: Configuration
+        for diagnostic in diagnostics {
+            let diag_point = MultiBufferPoint::new(diagnostic.range.start.row, 0);
+            let display_point = snapshot.point_to_display_point(diag_point, Bias::Right);
+
+            if !(start_row..end_row).contains(&display_point.row()) {
+                continue;
+            }
+            let line_ix = DisplayRow(display_point.row().minus(start_row));
+
+            if let Some(ref mut elem) = inline_diagnostics.get_mut(&line_ix) {
+                if elem.entries.len() >= max_entries {
+                    elem.exceeded_max = true;
+                    continue;
+                }
+
+                let entry = DiagnosticDisplay::new(
+                    diagnostic.diagnostic.severity,
+                    diagnostic.diagnostic.group_id,
+                    cx.new_view(|cx| {
+                        DiagnosticTooltip::new(diagnostic, hitbox, line_height, em_width, cx)
+                    }),
+                );
+                elem.entries.push(entry);
+                continue;
+            }
+
+            let start_y = content_origin.y
+                + line_height
+                    * (display_point.row().0 as f32 - scroll_pixel_position.y / line_height);
+
+            let start_x = {
+                const INLINE_BLAME_PADDING_EM_WIDTHS: f32 = 6.;
```

Comment:
It's not blame though, so we'd better rename or, ideally, have some common bits extracted into a function/method.

Reply:
This is now configurable and this padding constant is gone.